src/uo/packets.py

```diff
@@ -243,10 +243,10 @@ def __init__(self, serial, packet):
         self.item_id = packet.ushort()
         self.layer = packet.byte()
         if (self.item_id & 0x8000) != 0:
-            self.item_id &= ~0x8000
+            self.item_id &= 0x7FFF
             self.hue = packet.ushort()
         else:
-            self.hue = 0
+            self.hue = packet.ushort()
```

Comment:
This is wrong. The hue is only present if the highest `item_id` bit is set. See http://www.hoogi.de/wolfpack/wiki/doku.php?id=uo_protocol_0x78 or http://necrotoolz.sourceforge.net/kairpacketguide/packet78.htm or https://docs.polserver.com/packets/index.php?Packet=0x78

Reply:
Ok but if i leave these i get this error:
```
Traceback (most recent call last):
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/python/log.py", line 103, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/python/log.py", line 86, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
--- <exception caught here> ---
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/internet/selectreactor.py", line 149, in _doReadOrWrite
    why = getattr(selectable, method)()
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/internet/tcp.py", line 243, in doRead
    return self._dataReceived(data)
  File "/Users/mattiarossi/.pyenv/versions/gemuo/lib/python3.8/site-packages/twisted/internet/tcp.py", line 249, in _dataReceived
    rval = self.protocol.dataReceived(data)
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/client.py", line 78, in dataReceived
    self.on_packet(packet)
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/client.py", line 66, in on_packet
    self.handler(packet)
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/gemuo/client.py", line 48, in _handle_packet
    packet = p.parsers[packet.cmd](packet)
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/packets.py", line 264, in __init__
    self.items.append(MobileItem(serial, packet))
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/packets.py", line 248, in __init__
    self.item_id = packet.ushort()
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/serialize.py", line 89, in ushort
    return struct.unpack('>H', self.data(2))[0]
  File "/Users/mattiarossi/Develop/uo/gemuo2/src/uo/serialize.py", line 78, in data
    raise ProtocolError("Packet is too short")
uo.error.ProtocolError: Packet is too short
```

I take the modification from this handler https://github.com/andreakarasho/ClassicUO/blob/master/src/Network/PacketHandlers.cs#L2172 on ClassicUO project.

Reply:
Do you still make GemUO pretend it's talking a newer protocol version? That would explain your problems. I told you not to...

Reply:
Ok now i get it, you don't want support (for now) a new protocol version, right?